src/steps/pre-query/parse-request-primary.js

```diff
@@ -22,14 +22,24 @@ export default function(data, parseAsLinkage) {
     }
 
     catch (error) {
-      const title = "The resources you provided could not be parsed.";
-      const details = `The precise error was: "${error.message}".`;
-      reject(new APIError(400, undefined, title, details));
+      if (error instanceof APIError) {
+        reject(error);
+      }
+
+      else {
+        const title = "The resources you provided could not be parsed.";
+        const details = `The precise error was: "${error.message}".`;
+        reject(new APIError(400, undefined, title, details));
+      }
     }
   });
 }
 
-function relationshipObjectFromJSON(json) {
+function relationshipObjectFromJSON(json, relationship) {
+  if (typeof json.data === "undefined") {
+    throw new APIError(400, undefined, `No data key was found for the '${relationship}' relationship.`);
```

Comment:
> Are you able to explain some of the reasoning for that structure?

Short answer: the current structure is a mistake (for the reasons you cite) and I've already started to phase it out. Take a look at the http compliance tests, which use `before` and call `.done` on `Agent.request` promises to make sure failures aren't silent.

Feel free to write the new tests modeled after that, or, if that feels too inconsistent in a given file, to use the current structure but add `.done()`/`.catch()` calls on the promises to make sure errors get reported. Here's how that second approach would work (amending your first code sample):

```
describe((done) => {
  makeRequest().then((res) => {
    describe('Stuff', () => {
      it('should do something', () => {
        expect(true).to.be.ok;
      });
    });
  }).catch(done);
});
```